spec/classes/powerdns_init_spec.rb

```diff
@@ -632,40 +646,96 @@
           it { is_expected.to contain_package(authoritative_package_name).with('ensure' => 'installed') }
         end
 
-        context 'powerdns class with the recursor with forward zones' do
-          let(:params) do
-            {
-              recursor: true,
-              authoritative: false,
-              forward_zones: {
-                'example.com': '1.1.1.1',
-                '+.': '8.8.8.8'
+        if yaml_os?(facts)
+          # if a yaml OS, test the yaml style forward zones
+          context 'powerdns class with the recursor with yaml forward zones' do
```

Comment:
I think context blocks support conditions as well:
```suggestion
          context 'powerdns class with the recursor with yaml forward zones', if: yaml_os?(facts) do
```

That will save one level of indentation.

Reply:
Fixed